code/modules/overmap/ftl_shunt/core.dm

```diff
@@ -0,0 +1,551 @@
+	QDEL_IN(W, 6 SECONDS)
+	addtimer(CALLBACK(src, .proc/do_shunt, shunt_x, shunt_y, jumpdist, destination), 6 SECONDS)
+	jumping = TRUE
+	update_icon()
+	for(var/mob/living/carbon/M in world)
+		if(!(M.z in ftl_computer.linked.map_z))
+			continue
+		sound_to(M, 'sound/machines/hyperspace_begin.ogg')
+
+/obj/machinery/ftl_shunt/core/proc/do_shunt(var/turfx, var/turfy, var/jumpdist, var/destination) //this does the actual teleportation, execute_shunt is there to give us time to do our fancy effects
+	ftl_computer.linked.forceMove(destination)
+	ftl_announcement.Announce(shunt_complete_text, "FTL Shunt Management System", new_sound = sound('sound/misc/notice2.ogg'))
+	cooldown = world.time + cooldown_delay
+	update_use_power(POWER_USE_IDLE)
+	do_effects(jumpdist)
+	jumping = FALSE
+	update_icon()
+	chargepercent = 0
+	charge_started = 0
+
+//Handles all the effects of the jump.
+/obj/machinery/ftl_shunt/core/proc/do_effects(var/distance) //If we're jumping too far, have some !!FUN!! with people and ship systems.
+	var/shunt_sev
+	switch(distance)
+		if(1 to 3)
+			shunt_sev = SHUNT_SEVERITY_MINOR
+		if(4 to 5)
+			shunt_sev = SHUNT_SEVERITY_MAJOR
+		if(6 to INFINITY)
+			shunt_sev = SHUNT_SEVERITY_CRITICAL
+
+	for(var/mob/living/carbon/human/H in GLOB.living_mob_list_) //Affect mobs, skip synthetics.
+		sound_to(H, 'sound/machines/hyperspace_end.ogg')
+
+		if(!(H.z in ftl_computer.linked.map_z))
+			continue
+
+		handle_spacefloat(H)
+
+		if(isnull(H) || QDELETED(H))
+			continue
+
+		if(H.isSynthetic())
+			continue //We don't affect synthetics.
+
+		switch(shunt_sev)
+			if(SHUNT_SEVERITY_MINOR)
+				to_chat(H, SPAN_NOTICE("You feel your insides flutter about inside of you as you are briefly shunted into an alternate dimension.")) //No major effects.
+				shake_camera(H, 2, 1)
+
+			if(SHUNT_SEVERITY_MAJOR)
+				to_chat(H, SPAN_WARNING("You feel your insides twisted inside and out as you are violently shunted between dimensions, and you feel like something is watching you!"))
+				if(prob(25))
+					H.set_hallucination(50, 50)
+				if(prob(15))
+					H.vomit()
+				shake_camera(H, 2, 1)
+
+			if(SHUNT_SEVERITY_CRITICAL)
+				to_chat(H, SPAN_DANGER("You feel an overwhelming sense of nausea and vertigo wash over you, your instincts screaming that something is wrong!"))
+				if(prob(50))
+					H.set_hallucination(100, 100)
+				if(prob(45))
+					H.vomit()
+				shake_camera(H, 5, 4)
+
+	for(var/obj/machinery/light/L in SSmachines.machinery) //Fuck with and or break lights.
+		if(!(L.z in ftl_computer.linked.map_z))
+			continue
+		switch(shunt_sev)
+			if(SHUNT_SEVERITY_MINOR)
+				if(prob(15))
+					L.flicker()
+			if(SHUNT_SEVERITY_MAJOR)
+				if(prob(35))
+					L.flicker()
+
+	for(var/obj/machinery/power/apc/A in SSmachines.machinery)
+		if(!(A.z in ftl_computer.linked.map_z))
+			continue
+		switch(shunt_sev)
+			if(SHUNT_SEVERITY_MAJOR)
```

Comment:
This is not really in scope I guess, but I feel like you could datumize/declize the failure states to avoid this big chunky switch statements.